src/sinks/aws_cloudwatch_metrics/mod.rs

```diff
@@ -215,11 +215,11 @@ impl CloudWatchMetricsSvc {
     ) -> crate::Result<VectorSink> {
         let default_namespace = config.default_namespace.clone();
         let batch = config.batch.into_batch_settings()?;
-        let request_settings = config.request.unwrap_with(&TowerRequestConfig {
-            timeout_secs: Some(30),
-            rate_limit_num: Some(150),
-            ..Default::default()
-        });
+        let request_settings = config.request.unwrap_with(
+            &TowerRequestConfig::default()
+                .timeout_secs(30)
```

Comment:
Dropping the `Option` type from `timeout_secs` changes the behavior of this a bit. If the user configures `timeout_secs` to be the default explicitly (60), we'll still override the setting to 30 because we consider the default values to be cases where the user didn't provide configuration.

Reply:
Good catch. Tying my response to your other comment in with this one,  we actually need _something_ like the previous logic, to handle this ability to have three levels: user level config , sink specific default, and general default.

I think the ideal solution here is to convert this design into something like the `BatchConfig` settings, where each sink defines it's defaults in a struct. The overrides are built in and there is no need for that cumbersome unwrap_or function.
I am happy to refactor that.
I'll take a look at how much work it would be but, I may just revert the changes to remove the optionals that I have done here in this PR, because what I have introduces a regression. In the spirit of trying to accomplish the work on time.

Reply:
ok, I believe I've addressed this in the latest changes.
Ended up reverting to the optionals.
Looking at how BatchConfig is implemented, it seems do able for sure, but deserves a separate PR.